data-prepper-plugins/geoip-prepper/src/main/java/com/amazon/dataprepper/plugins/prepper/geoip/provider/LocationData.java

```diff
@@ -1,43 +1,83 @@
 package com.amazon.dataprepper.plugins.prepper.geoip.provider;
 
-
-// This is a first pass of LocationData, simply to make sure lookup correctly passes through data.
-//TODO Find the best way to attach the location data to a document, and modify this class to allow that.
-
+import java.util.HashMap;
+import java.util.Map;
 import java.util.Objects;
 
 public class LocationData {
+    private final String ip;
     private final String countryName;
     private final String subdivisionName;
     private final String cityName;
+    private final Map<String, Object> location;
+    private final Double latitude;
+    private final Double longitude;
+    private final String continentCode;
+    private final String countryCode;
+    private final String postalCode;
+    private final String regionCode;
+    private final String timezone;
 
-    public LocationData(String countryName, String subdivisionName, String cityName) {
-        this.countryName = countryName;
-        this.subdivisionName = subdivisionName;
-        this.cityName = cityName;
+    public LocationData(Map<Fields, Object> fields) {
+        ip = (String) fields.get(Fields.IP);
+        countryName = (String) fields.get(Fields.COUNTRY_NAME);
+        subdivisionName = (String) fields.get(Fields.REGION_NAME);
+        cityName = (String) fields.get(Fields.CITY_NAME);
+        Double[] latlon = (Double[]) fields.get(Fields.LOCATION);
+        continentCode = (String) fields.get(Fields.CONTINENT_CODE);
+        countryCode = (String) fields.get(Fields.COUNTRY_CODE2);
+        postalCode = (String) fields.get(Fields.POSTAL_CODE);
+        regionCode = (String) fields.get(Fields.REGION_CODE);
+        timezone = (String) fields.get(Fields.TIMEZONE);
+
+        if (latlon != null && latlon.length == 2) {
+            location = new HashMap<String, Object>() {{
+                put("lat", latlon[0]);
+                put("lon", latlon[1]);
+            }};
+        } else {
+            location = null;
+        }
+        latitude = (Double) fields.get(Fields.LATITUDE);
+        longitude = (Double) fields.get(Fields.LONGITUDE);
     }
 
-    @Override
-    public String toString() {
-        return "LocationData{" +
-                "countryName='" + countryName + '\'' +
-                ", subdivisionName='" + subdivisionName + '\'' +
-                ", cityName='" + cityName + '\'' +
-                '}';
+    public Map<String, Object> toMap() {
+        final Map<String, Object> mapping = new HashMap<>();
+        if (this.countryName != null)
```

Comment:
This is OK for now. But we should still consider jackson ObjectMapper combined with Jackson annotation on class attributes as a path to simplify if possible.

Reply:
+1, I'm not super familiar with JSON serializers but it'd be worth checking if it's just a matter of annotating the instance variables. Can be a separate PR though as @chenqi0805 mentioned.

Actually, doing this now would probably make the PR simpler, using an object mapper instead of a custom toMap method.